### PR TITLE
Wrong data inserted when COPY files to an altered partition table.

### PR DESCRIPTION
**Pull request by eurekaka**

### CODE REVIEW

**Comment by eurekaka** [comment]
@yydzero @xiong-gang @pengzhout any comments? thanks

**Comment by eurekaka** [comment]
@atris could you please help review? thanks

**Comment by yydzero** [comment]
Could you please add some description?

**Comment by eurekaka** [comment]
@yydzero the problem is: table tbl(col1, col2) has two partitions, e.g, part_a, part_b, then col1 of tbl has been dropped and then added back, so tbl would be like (col1/marked as droped, col2, clo1), then if we exchange a table exhg_tbl(col1, col2) with part_b, so now tbl(col1/deprecated, col2, col1) has partitions including part_a(col1/deprecated, col2, col1) and exhg_tbl(col1, col2), then if we copy a line of data into tbl which would be directed to exhg_tbl, wrong data would be inserted.

The root cause is: the partition table's tuple descriptor is different from root table's, so we should rebuild the tuple using the partition's tuple descriptor before inserting.

File: src/backend/commands/copy.c [hunk 1]
```diff
@@ -3886,8 +3896,8 @@ CopyFrom(CopyState cstate)
 				MemoryContextSwitchTo(GetPerTupleMemoryContext(estate));
 
 				/* Initialize all values for row to NULL */
-				MemSet(values, 0, num_phys_attrs * sizeof(Datum));
-				MemSet(nulls, true, num_phys_attrs * sizeof(bool));
+				MemSet(baseValues, 0, num_phys_attrs * sizeof(Datum));
+				MemSet(baseNulls, true, num_phys_attrs * sizeof(bool));
 				/* reset attribute pointers */
```
Comment (yydzero) [inline src/backend/commands/copy.c]: better to use palloc0

**Comment by xiong-gang** [comment]
LGTM

**Comment by eurekaka** [comment]
merged, thanks for the reviewing.

**Pull request closed without merging**